translation: Block-level elements (#112)
webdoky/content: files/uk/web/html/block-level_elements/index.md

---

> title: Елементи що утворюють блоки в HTML

Reviewer: ```suggestion
title: Елементи, що утворюють блоки в HTML
```

Reviewer: ```suggestion
title: Блокові елементи в HTML
```

slug: Web/HTML/Block-level_elements
tags:
  - Beginner
  - Development
  - Guide
  - HTML
  - HTML5
  - Web
---

> Розглянемо поняття елементів що утворюють блок **block-level elements** в порівняння з [елементами що вбудовані всередину рядків](/uk/docs/Web/HTML/Inline_elements) (inline-level elements).

Reviewer: Переклав ближче до тексту + замінив терміни на ті, до яких ми прийшли в чаті + вніс оригінальний варіант в дужках всередину тексту посилання
```suggestion
У цій статті ми розглянемо блокові елементи в HTML, і те, чим вони відрізняються від [рядкових елементів (inline-level elements)](/uk/docs/Web/HTML/Inline_elements).
```

> Історично в HTML усі елементи поділяються на ці дві категорії, де елементи що утворюють блок можна порівняти з параграфами у тексті. В той час, коли елементи що вбудовані всередину рядка візуалізуються через зовнішній вигляд декількох слів у параграфі. Такий розподіл елементів за зовнішнім виглядом у CSS ми називаємо [зображення за замовчанням](/uk/docs/Web/CSS/CSS_Flow_Layout) **flow layout**. Елементи що утворюють блок займають цім блоком весь горизонтальний вимір батьківського елементу **контейнеру**, та вертикальний вимір відповідний власному контенту елемента. 

Reviewer: Тут задалеко від тексту. В другому реченні оригіналу акцент був на тому, що тепер ці властивості (блок чи рядок), які раніше були притаманні самим HTML-елементам, тепер є частиною візуалізації, тому передаються через CSS (а саме за допомогою характеристик Flow Layout). Останнє речення переформулював трохи простіше + узгодив першу і другу його частину

```suggestion
Історично в HTML усі елементи поділяються на ці дві категорії, де елементи, що утворюють блок, можна порівняти з абзацами у тексті. В той час, коли елементи що, вбудовані всередину рядка, візуалізуються через зовнішній вигляд декількох слів у абзаці. Оскільки це характеристики візуальної подачі даних, зараз вони описуються за допомогою CSS в [потоковій розкладці](/uk/docs/Web/CSS/CSS_Flow_Layout) **flow layout**. Блокові елементи займають весь горизонтальний вимір батьківського елементу **контейнеру**, а їхній вертикальний вимір відповідає висоті вмісту елемента. 
```

> У браузері елементи що утворюють блок показуються як штабелі змісту.

Reviewer: Про нові рядки загубилось
```suggestion
Браузери зазвичай показують блокові елементи, виділяючи їх новими рядками до, і після елементу. Можна собі уявити ці елементи як штабелі блоків.
```

> > **Нотатка:** елемент що утворює блок завжди починається з нового рядка у тілі контенту, та займає всю можливу ширину, що обмежена батьківським елементом чи сторінкою.

Reviewer: Контент я зазвичай заміняю відповідником "вміст", але тут воно не в'яжеться, тому просто розставлю коми)
```suggestion
> **Нотатка:** елемент, який утворює блок, завжди починається з нового рядка у тілі контенту, та займає всю можливу ширину, що обмежена батьківським елементом чи сторінкою.
```

> Наведемо наочний приклад

Reviewer: Частина інформації загубилася
```suggestion
Наведений далі приклад ілюструє вплив блокових елементів
```

> ## Елемент що утворює блок

Reviewer: ```suggestion
## Блокові елементи
```

### HTML

```html

> <p>This paragraph is a block-level element; its background has been colored to display the paragraph's parent element.</p>

Reviewer: ```suggestion
<p>Цей абзац — блоковий елемент, і його тло забарвлено, щоб проявити батьківський елемент абзацу.</p>
```

```

### CSS

```css
p { background-color: #8ABB55; }
```

{{ EmbedLiveSample('Block-level_elements') }}

## Застосування

> - Елементи що утворюють блок можуть існувати виключно всередині елементу {{ HTMLElement("body") }} **тіло контенту**.

Reviewer: елемент body - це радше **тіло документу**, а контенту там може й не бути
```suggestion
- Елементи, що утворюють блок, можуть існувати виключно всередині елементу {{ HTMLElement("body") }}.
```

> ## Специфіка HTML5

Reviewer: ```suggestion
## Порівняння блокових та рядкових елементів
```

> 

Reviewer: Теж трохи загубилося
```suggestion
Існує пара ключових відмінностей між блоковими і рядковими елементами:

- Модель вмісту
  - : Загалом, блокові елементи можуть містити рядкові елементи та (інколи) інші блокові елементи. Суть цієї структурної відмінності в тому, що блокові елементи створюють "більші" структури, ніж рядкові.
- Усталене форматування
  - : Усталено, блокові елементи починаються з нового рядка, а рядкові елементи можуть початися в будь-якому місці рядка.
```

> В HTML5 історична бінарна класифікація елементів заміщена на [категорії контенту](/uk/docs/Web/Guide/HTML/Content_categories). Серед них [фразова категорія контенту](/uk/docs/Web/Guide/HTML/Content_categories#phrasing_content) відповідає елементам що вбудовані всередину рядків. Логічно, що ці рядки є частиною елементів що утворюють блок, тобто все попадає у [категорію зображення за замовчанням](/en-US/docs/Web/Guide/HTML/Content_categories#flow_content) **flow layout**. З'явилися додаткові категорії контенту, такі як [інтерактивний контент](/uk/docs/Web/Guide/HTML/Content_categories#interactive_content) тощо.

Reviewer: Трохи загубилася конкретика, переформулював ближче до тексту
```suggestion
Розмежування між блоковими та рядковими елементами вказувалась у специфікації HTML аж до версії 4.01. В HTML5 історична бінарна класифікація елементів заміщена на складнішу — набір [категорій вмісту](/uk/docs/Web/Guide/HTML/Content_categories). В той час як категорія рядкових елементів приблизно відповідає [фразовому вмісту](/uk/docs/Web/Guide/HTML/Content_categories#phrasing_content), блокові елементи не мають прямого відповідника серед категорій вмісту HTML5. Проте _комбінація з рядкових і блокових елементів_ відповідає категорії [потокової розкладки](/uk/docs/Web/Guide/HTML/Content_categories#flow_content). З'явилися додаткові категорії контенту, такі як [інтерактивний контент](/uk/docs/Web/Guide/HTML/Content_categories#interactive_content) тощо.
```

> ## Перелік елементів що утворюють блок

Reviewer: Насправді це слушне уточнення, проте я не хочу сильно відхилятись від оригіналу. Якщо є бажання таки виправити щось - раджу спершу внести цю зміну в оригінальний репозиторій. Я сам так час від часу роблю, народ там доволі доброзичливий https://github.com/mdn/content/pulls?q=is%3Apr+author%3AAdriandeCita+is%3Aclosed
```suggestion
## Перелік елементів
```

> Нижче ми наводимо повний перелік елементів що утворюють блок у HTML, попре що в HTML5 ця класифікація не введена для нових категорій контенту.

Reviewer: Там маються на увазі не нові категорії, а нові елементи
```suggestion
Нижче ми наводимо повний перелік блокових елементів у HTML, (попри те, що ця категорія не визначена для нових елементів, які з'явились в HTML5).
```

- {{ HTMLElement("address") }}

>   - : контактна інформація

Reviewer: ```suggestion
  - : Контактна інформація
```

- {{ HTMLElement("article") }}

>   - : контент статті

Reviewer: ```suggestion
  - : Контент статті
```

- {{ HTMLElement("aside") }}

>   - : додатковий контент

Reviewer: ```suggestion
  - : Додатковий контент
```

- {{ HTMLElement("blockquote") }}

>   - : цитата

Reviewer: ```suggestion
  - : Довга цитата (блок)
```

- {{ HTMLElement("details") }}

>   - : віджет що надає доступ до довідки

Reviewer: ```suggestion
  - : Віджет, що розкривається
```

- {{ HTMLElement("dialog") }}

>   - : комунікаційна форма

Reviewer: Традиційно це називається терміном "діалогове вікно"
```suggestion
  - : Діалогове вікно
```

- {{ HTMLElement("dl") }}

>   - : перелік термінів

Reviewer: ```suggestion
  - : Перелік термінів
```

- {{ HTMLElement("dt") }}

>   - : термін у переліку термінів

Reviewer: ```suggestion
  - : Термін у переліку термінів
```

- {{ HTMLElement("dd") }}

>   - : значення терміну у переліку термінів

Reviewer: ```suggestion
  - : Опис терміну у переліку термінів
```

- {{ HTMLElement("div") }}

>   - : розділювач документу

Reviewer: Тут мається на увазі елемент, який містить частину документу
```suggestion
  - : Розділ документу
```

- {{ HTMLElement("fieldset") }}

>   - : елемент що групує інші елементи у спільному полі

Reviewer: Тут маються на увазі текстові поля вводу
```suggestion
  - : Елемент для групування полів вводу
```

- {{ HTMLElement("figcaption") }}

>   - : опис до медіаконтенту

Reviewer: ```suggestion
  - : Опис до медіаконтенту
```

- {{ HTMLElement("figure") }}

>   - : групує медіаконтент з описом до нього

Reviewer: ```suggestion
  - : Групує медіаконтент з описом до нього
```

- {{ HTMLElement("footer") }}

>   - : нижня секція сторінки

Reviewer: Там малося на увазі, що футер може бути як на сторінці загалом, так і на окремій статті. 
```suggestion
  - : Нижній колонтитул розділу або сторінки
```

- {{ HTMLElement("form") }}

>   - : форма для вводу даних

Reviewer: ```suggestion
  - : Форма для вводу даних
```

- {{ HTMLElement("h1") }}, {{ HTMLElement("h2") }}, {{ HTMLElement("h3") }}, {{ HTMLElement("h4") }}, {{ HTMLElement("h5") }}, {{ HTMLElement("h6") }}

>   - : рівні заголовків 1-6

Reviewer: ```suggestion
  - : Рівні заголовків 1-6
```

- {{ HTMLElement("header") }}

>   - : верхня секція сторінки

Reviewer: Те саме, там в оригіналі "Section OR page...". Для верхнього колонтитулу в українській є відповідник, але для консистентності я вказав також і типографічний термін.
```suggestion
  - : Верхній колонтитул (шапка) розділу чи сторінки
```

- {{ HTMLElement("hgroup") }}

>   - : Groups header information.

Reviewer: Пропущений переклад
```suggestion
  - : Об'єднує декілька елементів у заголовок.
```

- {{ HTMLElement("hr") }}

>   - : семантична горизонтальна лінія

Reviewer: Всі елементи мають певну семантику, виділяти hr додатково не потрібно
```suggestion
  - : Горизонтальна лінія (роздільник)
```

- {{ HTMLElement("li") }}

>   - : перелік

Reviewer: Для власне переліку існує інший елемент
```suggestion
  - : Елемент переліку
```

- {{ HTMLElement("main") }}

>   - : головна сторінка

Reviewer: ```suggestion
  - : Містить головний зміст, унікальний для цього документу
```

- {{ HTMLElement("nav") }}

>   - : посилання для навігації

Reviewer: ```suggestion
  - : Контейнер для навігаційних посилань
```

- {{ HTMLElement("ol") }}

>   - : впорядкований перелік

Reviewer: ```suggestion
  - : Впорядкований перелік
```

- {{ HTMLElement("p") }}

>   - : параграф

Reviewer: Так, це спокусливо, але насправді параграф у нас - це радше цілий розділ) А правильний переклад - абзац
```suggestion
  - : параграф
```

- {{ HTMLElement("pre") }}

>   - : текст з попереднім форматуванням

Reviewer: ```suggestion
  - : Текст з попереднім форматуванням
```

- {{ HTMLElement("section") }}

>   - : секція на сторінці

Reviewer: ```suggestion
  - : Секція на сторінці
```

- {{ HTMLElement("table") }}

>   - : таблиця

Reviewer: ```suggestion
  - : Таблиця
```

- {{ HTMLElement("ul") }}

>   - : невпорядкований перелік

Reviewer: ```suggestion
  - : Невпорядкований перелік
```

> ## Також дивіться

Reviewer: Для консистентності з іншими перекладами
```suggestion
## Дивіться також
```

> - [елементи що вбудовані всередину рядків](/uk/docs/Web/HTML/Inline_elements)

Reviewer: ```suggestion
- [Рядкові елементи](/uk/docs/Web/HTML/Inline_elements)
```

- {{cssxref("display")}}

> - [елементи що утворюють блоки та що вбудовані всередину рядків як зображення за замовчанням](/uk/docs/Web/CSS/CSS_Flow_Layout/Block_and_Inline_Layout_in_Normal_Flow)

Reviewer: Про зображення - це трохи зайве)
```suggestion
- [Розміщення рядкових та блокових елементів у звичайній потоковій розкладці](/uk/docs/Web/CSS/CSS_Flow_Layout/Block_and_Inline_Layout_in_Normal_Flow)
```
